packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -140,7 +297,337 @@ class FastILUPrec
         using STS = Kokkos::ArithTraits<Scalar>;
         using RTS = Kokkos::ArithTraits<Real>;
 
-    private:
+  struct IdentityFunctor
```

Comment:
A bunch of functors for when blockcrs has to do different thing than regular crs.